tests/test_config.py

```diff
@@ -29,10 +28,22 @@ def autouse_config_tmpdir(config_tmpdir):
 
 def cfg(f=ALL, topdir=None):
     # Load a fresh configuration object at the given level, and return it.
-    cp = configparser.ConfigParser(allow_no_value=True)
-    config.read_config(configfile=f, config=cp, topdir=topdir)
+    cp = config.configparser.ConfigParser(allow_no_value=True)
```

Comment:
Mostly not your fault but this commit is incredibly hard to review... first, this import rename is horrible because it hides both the origin and nature of `config`:
```python
from west import configuration as config
```
(don't change it now because it would have a massive ripple effect)

Then, I can't find `west.configuration.configparser` in the documentation. Is this is some "private" object (ab)used for "white box" testing? If yes please add a comment stating that. If not, sorry I missed what it is.

The icing on the cake is the fact that this commit replaces Python's built-in `configparser` with `west.configuration.configparser` which has the very same name while being a bit of a mystery...

I hate to ask but... could you slow down a little bit? Maybe move this to a separate commit. Either way, provide a bit more comments, commit message, etc. Thanks in advance!

Reply:
I was contemplating the same things, but tried to touch it "minimally" I'll try to clean this up so that it makes sense what is builtin and what not.

Reply:
I've reverted to the original `configparser`.